pvlib/iotools/ecmwf_macc.py

```diff
@@ -0,0 +1,226 @@
+"""
+Read data from ECMWF MACC Reanalysis.
+"""
+
+from __future__ import division
+import threading
+import pandas as pd
+
+try:
+    import netCDF4
+except ImportError:
+    class netCDF4:
+        def Dataset(*a, **kw):
+            raise ImportError(
+                'Reading ECMWF data requires netCDF4 to be installed.')
+
+try:
+    from ecmwfapi import ECMWFDataServer
+except ImportError:
+    def ECMWFDataServer(*a, **kw):
+        raise ImportError(
+            'To download data from ECMWF requires the API client.\nSee https:/'
+            '/confluence.ecmwf.int/display/WEBAPI/Access+ECMWF+Public+Datasets'
+        )
+    SERVER = None
+else:
+    SERVER = ECMWFDataServer()
+
+PARAMS = {
+    "tcwv": "137.128",
+    "aod550": "207.210",
+    'aod469': '213.210',
+    'aod670': '214.210',
+    'aod865': '215.210',
+    "aod1240": "216.210",
+}
+
+
+def _ecmwf(server, startdate, stopdate, params, targetname):
+    # see http://apps.ecmwf.int/datasets/data/macc-reanalysis/levtype=sfc/
+    server.retrieve({
+        "class": "mc",
+        "dataset": "macc",
+        "date": "%s/to/%s" % (startdate, stopdate),
+        "expver": "rean",
+        "grid": "0.75/0.75",
+        "levtype": "sfc",
+        "param": params,
+        "step": "3/6/9/12/15/18/21/24",
+        "stream": "oper",
+        "format": "netcdf",
+        "time": "00:00:00",
+        "type": "fc",
+        "target": targetname,
+    })
+
+
+def get_ecmwf_macc(filename, params, startdate, stopdate, lookup_params=True,
+                   server=SERVER, target=_ecmwf):
+    """
+    Download data from ECMWF MACC Reanalysis API.
+
+    Parameters
+    ----------
+    filename : str
+        full path of file where to save data, ``.nc`` appended if not given
+    params : str or sequence of str
+        keynames of parameter[s] to download
+    startdate : datetime.datetime or datetime.date
+        UTC date
+    stopdate : datetime.datetime or datetime.date
+        UTC date
+    lookup_params : bool
+        optional flag, if ``False``, then codes are already formatted, default
+        is ``True``
```

Comment:
while not yet a hard rule, it's more common in pvlib to see `lookup_params : bool, default True`. The description only needs to mention the default if it needs to be explained.